### DISCUSSION TITLE
Failed branches because of cancelled/broken CI runs do not get PR created

### DISCUSSION THREAD

**Discussion by janpio** [renovatebot/renovate#10065] [Request Help]
(Hosted Github app)

During the GH Actions outage we killed a lot of our CI runs or they failed on Github directly, causing red branches. 
It seems that Renovate is not opening PRs for these branches for some reason.

![image](https://user-images.githubusercontent.com/183673/118938791-4e2c1980-b94f-11eb-885f-57613a450a31.png)
https://github.com/prisma/e2e-tests/issues/61

https://github.com/prisma/e2e-tests/tree/renovate/serverless-2.x
https://github.com/prisma/e2e-tests/actions/runs/858537180

I think I could now fix this by clicking the `Check this option to rebase all the above open PRs at once` in the dependency issue (or even just restart my CI), as that would rerun the tests and the probably merge. But as I am not in a hurry, I thought it might be useful to report this to you so you can investigate.

Problem might be that Renovate thinks these branches are pending, from the last log:

```
DEBUG: branch status check result(branch="renovate/serverless-2.x")
{
  "state": "pending",
  "statuses": []
}
DEBUG: check runs result(branch="renovate/serverless-2.x")
```

**Comment by viceice** [comment]
🤔 the branches are outdated, are they maked for automerge? If not, renovate waits for 25h before opening pr for pending branches
![image](https://user-images.githubusercontent.com/1798109/118941433-022ea400-b952-11eb-9b63-f131d62db012.png)


https://docs.renovatebot.com/configuration-options/#prnotpendinghours

Can you check the [app dashbord](https://app.renovatebot.com/dashboard) when last run was happen? and also post the logs here?

**Reply by rarkins** [reply]
Was there another actions outage that recently?

**Reply by janpio** [reply]
GH Actions was down last weekend (Friday to Saturday I think), then Tuesday and again spotty yesterday evening (CET).

https://app.renovatebot.com/dashboard#github/prisma/e2e-tests
![image](https://user-images.githubusercontent.com/183673/118954858-75d6ae00-b95e-11eb-8c6c-5eced4029c38.png)

Here is the complete log of the last run:
[renovate.log](https://github.com/renovatebot/renovate/files/6514636/renovate.log)

**Comment by rarkins** [comment]
Thanks for reporting it.

The dependency dashboard unfortunately falls victim to some edge cases and thinks things are "Open" when they're not. I'll treat that as a separate problem for now.

I see in your repo you have it configured for branch automerge: https://github.com/prisma/e2e-tests/blob/14758fbc6ee3b40f4ac727282e0711370142607e/renovate.json#L4-L5

And the branches are "pending" because no tests got started.

We do have a `prNotPendingHours` timeout for the prCreation=not-pending setting to prevent a similar situation but maybe for automergeType=branch such a check isn't done.

**Reply by janpio** [reply]
Sounds good. Do you want to investigate and change something here or should I just go ahead and restart tests?

**Reply by viceice** [reply]
I've checked your logs, so i think renvate has already updated some branches and will do automerge on next run. So you will slowly see self-healing.

because of the automerge every second run, renovate breaks early and don't update the dashbord or other branches.

@rarkins Maybe we can reorder branch processing to check if any branch is ready for automerge and do this first, so no rebasings before automerge, as this will nessessary again on next run. 🤔
